pkg/auth/token_verifier.go

```diff
@@ -179,3 +179,19 @@ type openIDMetadata struct {
 	SubjectTypes  []string `json:"subject_types_supported"`
 	SigningAlgs   []string `json:"id_token_signing_alg_values_supported"`
 }
+
+// Getting the OIDCIdentity
+func (c *OIDCTokenVerifier) GetOIDCIdentity(ctx context.Context, r *http.Request, audience string) (*IDToken, error) {
```

Comment:
As mentioned in the other comment, the goal if your function is to get OIDC Identity and not verify the request as you are doing on line 191.